clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -334,14 +334,15 @@ public static void sleep(long ms) {
      * Instantiate the class
      */
     public static <T> T newInstance(Class<T> c) {
+        if (c == null)
+            throw new NullPointerException();
```

Comment:
Why are we changing the exception type here? That could affect calling code. Also, if there's a good reason to do this, we'd want to include a message in the exception.